cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -409,18 +412,30 @@ def __pow__(self, exponent) -> 'CliffordGate':
             )
         if exponent == 1:
             return self
-        if exponent > 0 and int(exponent) == exponent:
-            base_tableau = self.clifford_tableau.copy()
-            for _ in range(int(exponent) - 1):
-                base_tableau = base_tableau.then(self.clifford_tableau)
-            return CliffordGate.from_clifford_tableau(base_tableau)
-        if exponent < 0 and int(exponent) == exponent:
-            base_tableau = self.clifford_tableau.copy()
-            for _ in range(int(-exponent) - 1):
-                base_tableau = base_tableau.then(self.clifford_tableau)
-            return CliffordGate.from_clifford_tableau(base_tableau.inverse())
+        base_tableau = self.clifford_tableau.copy()
+        if exponent < 0:
+            base_tableau = base_tableau.inverse()
+            exponent = int(abs(exponent))
+
+        # https://cp-algorithms.com/algebra/binary-exp.html
+        operation_order: List[bool] = list()
```

Comment:
this is not binary exponenitation e.g. if the exponent is 3 this code will evaluate to self**4 instead of self**3.

try this instead

```py3
aux = cirq.qis.CliffordTableau(num_qubits=1)  # this tableau collects the odd terms, intially = Identity
while exponent > 1:
   if exponent & 1: aux = aux.then(base_tableau)
   base_tableau = base_tableau.then(base_tableau)
   exponent >>= 1

base_tableau = base_tableau.then(aux)
return CliffordGate.from_clifford_tableau(base_tableau)
```

Reply:
oh, I missed the `operation_order = operation_order[::-1]` line.

however this is still a bit convoluted, please prefer a simpler implementation
---
please refer to docs/dev/development.md for creating a dev environmental. personally I create a virtual env and then `pip install -r dev_tools/requirements/dev.env.txt`